### PR TITLE
LPS-96569 Provide top-level ESLint configuration

### PR DESCRIPTION
**Pull request by wincent**
This will allow editors such as VSCode to pick up the ESLint config and show lint problems inline in the IDE.

Unlike the Prettier config (which is static), our ESLint config depends on installed code (because some of the rules are defined in plugins that live inside node_modules). This required some refactoring, the details of which are all described in:

https://github.com/liferay/liferay-npm-tools/pull/130

The short version is that we moved the `root: true` config from this repo into liferay-npm-scripts, we grab the base configuration using `require()`, and if the user hasn't run "ant setup-sdk" (or "ant all") yet, we throw a helpful error message that IDEs will show.

Example from VSCode:

error

### CODE REVIEW

**Comment by wincent** [comment]
And here is a screenshot showing it actually working after an `ant setup-sdk`:

good

**Comment by brianchandotcom** [comment]
@hhuijser fyi.
